dbms/src/Storages/Page/V3/WALStore.cpp

```diff
@@ -201,7 +201,11 @@ bool WALStore::saveSnapshot(FilesSnapshot && files_snap, PageEntriesEdit && dire
     // Rename it to be a normal log file.
     const auto temp_fullname = log_filename.fullname(LogFileStage::Temporary);
     const auto normal_fullname = log_filename.fullname(LogFileStage::Normal);
+
     LOG_FMT_INFO(logger, "Renaming log file to be normal [fullname={}]", temp_fullname);
+    // Need link encryption info before rename
+    provider->linkEncryptionInfo(temp_fullname, normal_fullname);
```

Comment:
Should also remove the encryption info for temp_fullname after we done rename the file